### ISSUE TITLE
False-negative `signature-differs` with kw-only args

### ISSUE THREAD

**Issue by cdce8p** [PyCQA/pylint#5270]
### Bug description

The following example should emit `signature-differs` due to different default values.

```py
class Parent:
    def statement(self, *, future = None):
        pass

class Child(Parent):
    def statement(self, *, future):  # should emit 'signature-differs'
        pass
```

Removing `*` will yield:
```
W0222: Signature differs from overridden 'statement' method (signature-differs)
```

### Command used

```shell
pylint test.py
```

### Pylint output

```shell
-
```

### Expected behavior

W0222: Signature differs from overridden 'statement' method (signature-differs)

### Pylint version

```shell
pylint 2.11.2-dev0
astroid 2.8.5-dev0
Python 3.10.0 (v3.10.0:b494f5935c, Oct  4 2021, 14:59:20) [Clang 12.0.5 (clang-1205.0.22.11)]
```
